### PR TITLE
Implement Certificate Revocation List (2)

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

**Comment by Danielius1922** [comment]
@coderabbitai review

**Comment by Danielius1922** [comment]
@coderabbitai review

**Comment by Danielius1922** [comment]
@coderabbitai review

**Comment by Danielius1922** [comment]
@coderabbitai in the pull request walkthrough ignore the propagation of trace.TracerProvider to various functions, what is important in the code changes is the refactoring of code in pkg/security/certManager and that http client was integrated into the certificate manager types. That is why all other changes in this PR occured, so focus on that.

**Comment by Danielius1922** [comment]
@coderabbitai review

**Comment by Danielius1922** [comment]
@coderabbitai review

File: pkg/security/certManager/general/certManager.go [hunk 1]
```diff
@@ -78,17 +86,27 @@ func tryToWatchFile(file urischeme.URIScheme, fileWatcher *fsnotify.Watcher, rem
...
 	}
 
+	var httpClient *pkgHttpClient.Client
+	if config.CRL.Enabled {
+		var err error
+		httpClient, err = NewHTTPClient(config.CRL.HTTP, fileWatcher, logger, tracerProvider)
+		if err != nil {
+			return nil, err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Danielius1922) [reply pkg/security/certManager/general/certManager.go]: Good catch, thanks

**Review by jkralik — APPROVED** [review]

**Comment by Danielius1922** [comment]
@coderabbitai review

File: pkg/security/tls/crl.go [hunk 2]
```diff
@@ -71,3 +83,83 @@ func (c *Config) Validate() error {
...
+		return false
+	}
+	tls := c.HTTP.GetTLS()
+	return c.HTTP.GetMaxIdleConns() == c2.HTTP.GetMaxIdleConns() &&
+		c.HTTP.GetMaxConnsPerHost() == c2.HTTP.GetMaxConnsPerHost() &&
+		c.HTTP.GetMaxIdleConnsPerHost() == c2.HTTP.GetMaxIdleConnsPerHost() &&
+		c.HTTP.GetIdleConnTimeout() == c2.HTTP.GetIdleConnTimeout() &&
+		c.HTTP.GetTimeout() == c2.HTTP.GetTimeout() &&
+		tls.Equals(c2.HTTP.GetTLS())
+}
```
[the remark this answers is not in the document]
Comment (jkralik) [reply pkg/security/tls/crl.go:136]: @Danielius1922  I think this is a good suggestion.

**Pull request merged**